script.js

```diff
@@ -195,4 +195,43 @@ form.addEventListener('submit', (event) => {
   } else {
     form.submit();
   }
-});
\ No newline at end of file
+});
+
+// ######## Local storage ########
+
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return (
+      (e instanceof DOMException && e.code === 22)
+      || e.code === 1014 || e.name === 'QuotaExceededError'
+      || (e.name === 'NS_ERROR_DOM_QUOTA_REACHED' && storage && storage.length !== 0)
+    );
+  }
+}
+
+if (storageAvailable('localStorage')) {
+  const inputs = [form.name, form.email, form.message];
+  inputs.forEach((input) => {
+    input.addEventListener('input', () => {
+      const storagData = {
+        fullName: form.name.value,
+        email: form.email.value,
+        message: form.message.value,
+      };
+      localStorage.setItem('data', JSON.stringify(storagData));
+    });
+  });
+  const getData = JSON.parse(localStorage.getItem('data'));
+  if (getData) {
+    form.name.value = getData.name;
+    form.email.value = getData.email;
+    form.message.value = getData.message;
+  }
+}
```

Comment:
There is a tiny bug with your code at the moment. Your email and text are being saved and the persistence is working for them but not for the name. As you can see in the following image, the name text is changed to undefined. 

![image](https://user-images.githubusercontent.com/15807343/162336954-df156fff-4d19-46dd-a1c5-9fe401f2d289.png)

Kindly make the necessary changes to solve the issue described 🧐

Reply:
we did it